controllers/cloud.redhat.com/clowdenvironment_controller.go

```diff
@@ -177,12 +183,86 @@ func runProvidersForEnv(provider providers.Provider) error {
 func (r *ClowdEnvironmentReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	r.Recorder = mgr.GetEventRecorderFor("env")
 	return ctrl.NewControllerManagedBy(mgr).
-		Owns(&apps.Deployment{}).
-		Owns(&core.Service{}).
 		For(&crd.ClowdEnvironment{}).
+		Watches(
+			&source.Kind{Type: &crd.ClowdApp{}},
+			&handler.EnqueueRequestsFromMapFunc{
+				ToRequests: handler.ToRequestsFunc(r.envToEnqueueUponAppUpdate),
+			},
+		).
 		Complete(r)
 }
 
+func (r *ClowdEnvironmentReconciler) envToEnqueueUponAppUpdate(a handler.MapObject) []reconcile.Request {
+	ctx := context.Background()
+	obj := types.NamespacedName{
+		Name:      a.Meta.GetName(),
+		Namespace: a.Meta.GetNamespace(),
+	}
+
+	// Get the ClowdEnvironment resource
+
+	app := crd.ClowdApp{}
+	err := r.Client.Get(ctx, obj, &app)
+
+	if err != nil {
+		if k8serr.IsNotFound(err) {
+			// Must have been deleted
+			return []reconcile.Request{}
+		}
+		r.Log.Error(err, "Failed to fetch ClowdApp")
+		return nil
+	}
+
+	return []reconcile.Request{{
+		NamespacedName: types.NamespacedName{
+			Name: app.Spec.EnvName,
+		},
+	}}
+}
+
+func (r *ClowdEnvironmentReconciler) generateAppStatus(p providers.Provider) error {
+	// Get all the ClowdApp resources
+	appList := crd.ClowdAppList{}
+	r.Client.List(p.Ctx, &appList)
+	apps := []crd.AppStatus{}
+
+	// Populate
+	for _, app := range appList.Items {
+
+		if app.Spec.Pods != nil {
+			app.ConvertToNewShim()
+		}
+
+		appstatus := crd.AppStatus{
+			Name:        app.Name,
+			Deployments: []crd.DeploymentStatus{},
+		}
+
+		if app.Spec.EnvName == p.Env.Name {
```

Comment:
imho it would be better to change this to 

```suggestion
		if app.Spec.EnvName != p.Env.Name {
		    continue
		}
```

and move it up to the top of the loop

Reply:
The reasons are 1) fewer indented lines, and 2) you don't create structs that get thrown away.  pretty minor stuff